### ISSUE TITLE
Simplify selected feature deletion

### ISSUE THREAD

**Issue by yaugenka** [openlayers/ol3#4214]
To delete selected features you need to find the layer they belong to first:

    var select = new ol.interaction.Select();
    select.getFeatures().forEach(function(feature){
       layer.getSource().removeFeature(feature);
    });

This gets quite hard if there are several layers especially grouped ones. Take a look at this [post](http://gis.stackexchange.com/questions/164229/remove-or-delete-selected-feature-in-openlayers).
It would be good to make it simpler somehow.

**Comment by ahocevar** [comment]
You are right, this is not ideal. Especially since the layer is known when a feature is selected. An improved API could look like this:
```js
selectInteraction.getLayer(selectedFeature).getSource().removeFeature(selectedFeature);
```
Would you be able/willing to create a pull request for this? Internally, all the Select interaction would need to provide such an API would be an index of features by layer, using `goog.getUid(feature)` for the index keys and `layer` as value.

**Comment by yaugenka** [comment]
I don't think I'll manage it yet because i started using html/javascipt just a couple of weeks ago :)

**Comment by jonataswalker** [comment]
@ahocevar I think I can handle this PR but wouldn't a ol.Feature.getLayer() more useful? I mean a broader method?

**Comment by jonataswalker** [comment]
Also, the known layer to `ol.interaction.Select` is not the layer we want, it is this internal `this.featureOverlay_`. Isn't it?

**Comment by alvinlindstam** [comment]
Wouldn't an issue with @ahocevar's or @jonataswalkers suggestions be the possibility that a feature may be used in several layers/sources?

**Comment by ahocevar** [comment]
Exactly @alvinlindstam. This is why we don't want a layer property on ol.Feature. But the moment a feature is selected, it is selected from one layer, and the Select interaction can record that layer for later use because it is known.

**Comment by ahocevar** [comment]
> Also, the known layer to ol.interaction.Select is not the layer we want, it is this internal this.featureOverlay_. Isn't it?

No. The place to record layers for later use would be the function we pass to map.forEachFeatureAtPixel(). The layer passed to that function is the original layer, not the selection overlay.

**Comment by fredj** [comment]
Fixed with #4218 and #4370
